test/parallel/test-assert-deep.js

```diff
@@ -466,4 +466,10 @@ assertOnlyDeepEqual(
   assertDeepAndStrictEqual(m3, m4);
 }
 
+// Handle NaN
+{
+  assert.throws(() => assert.deepEqual(NaN, NaN), assert.AssertionError);
```

Comment:
Nit: Leaving out the braces around the arrow function adds an implicit return. Since the returned value is not used, I prefer to add braces. Feel free to ignore, but my two cents on it.